documentation/specs/proposed/BuildCheck.md

```diff
@@ -0,0 +1,405 @@
+* Custom anlyzer in a local project (source codes) or a binary.
+* Bulk configuration of multiple rules (based on prefixes).
+* Specifying scope of MSBuild imports that will be considered for analysis (so that e.g. data from sdk won't even be passed to anlyzer, if not requested).
+* Attempts to try to configure standard msbuild warnings/errors via `.editorconfig` should lead to fail fast errors.
+* Configuring analysis levels when analysing from binlog - beyond the collected editorconfigs
+* Rich information in VS error window.
+
+
+**Out of scope**
+* Custom analyzers has equal data access as the inbox analyzers.
+* All inbox analyzers reports have precise location(s) of issues.
+* Opt-out of analysis on code-level (analogy to C# pragmas, but within msbuild xml files).
+* Simplified authoring experience via dedicated reference assembly.
+* Restore phase analysis.
+* Turning analysis off/on based on target (e.g. multi-targetted builds, calling MSBuild task etc.).
+* Controling/Configuring lifetime of analyzers - analyzers will currently be held alive, as single instance per analyzer, for the whole duration of the build. But future versions might prevent some of the analyzers to survive beyond a scope of a single project built (means for sharing data would be provided).
+* ETW for analyzers.
+* Attributing `.editorconfig` configurations to .sln files.
+* Attributing `.editorconfig` configurations to lower granularity than whole projects.
+* Respecting `.editorconfig` file in msbuild import locations (unless they are in the parent folders hieararchy of particular project file).
+ 
+
+# User Experience
+
+## Running / UX
+
+### Inbox Analyzers
+
+**TBD** - suggested list of analyzers to be shipped with V1 (along with proposed configuration)
+
+### Live Build
+
+BuildCheck will run as part of the build and execute [inbox analyzers](#inbox-analyzers) and [custom analyzers](#acquisition-of-custom-analyzers) based on the [configuration](#configuration). Users will have an option to completely opt-out from BuildCheck to run via commandline switch.
+
+Findings - reports - of analyzers will be otput as build messages/warnings/errors, while the code should help distinguish BuildCheck produced reports from regular build errors/warnings.
+
+### Binlog Replay mode
+
+Users will have option to explicitly opt-in to run BuildCheck during the binlog replay mode:
+
+```bash
+> dotnet build msbuild.binlog /analyze
+```
+
+Would there be any analyzers that are not possible to run during the replay mode (subject to internal design - this difference won't be exposed during [custom analyzers authoring](#custom-analyzers-authoring)), replay mode will inform user about those via warnings.
+
+Replay mode will by default consider `.editorconfig` files stored within the binlog and will run analyzers based on those. This may lead to double-reports – as binlog will have the runtime analysis reports stored, plus the replay-time analysis reports will be augumented.
```

Comment:
Is the double reporting intentional? Feels like it could be a won't-fix bug for v1 but doesn't seem like something we intentionally want.

Reply:
My wording obviously needs some tweaking here :-)

As the message I wanted to transfer was what you pointed out - "not intended, but possibly not high priority for addressing in v1"